components/compositing/constellation.rs

```diff
@@ -638,17 +638,11 @@ impl<LTF: LayoutTaskFactory, STF: ScriptTaskFactory> Constellation<LTF, STF> {
 
         self.close_pipeline(pipeline_id, ExitPipelineMode::Force);
 
-        loop {
```

Comment:
We should do this as 

```rust
for frame in self.pending_frames.iter().filter(|pending| pending.old_pipeline_id == Some(pipeline_id)) {
  debug!("removing pending frame change for failed pipeline");
  self.close_pipeline(frame.new_pipeline_id, ExitPipelineMode::Force);
}
```

The loop can be replaced by a `.map(...).collect()` as well, though neither is preferred,